### PR TITLE
tests: add tests for grpc services

### LINKED ISSUE #2058: grpc_service's coverage is too low
ref: https://coveralls.io/builds/12454931/source?filename=src%2Fserver%2Fgrpc_service.rs

### PR DESCRIPTION
**Pull request by overvenus**
Close #2058

### CODE REVIEW

File: tests/raftstore/test_service.rs [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+    ctx.set_peer(leader.clone());
+    ctx.set_region_epoch(epoch);
+
+    let addr = cluster.sim.rl().get_addr(leader.get_store_id());
+    let env = Arc::new(Environment::new(1));
+    let channel = ChannelBuilder::new(env).connect(&format!("{}", addr));
+    let client = TikvClient::new(channel);
+    let (k, v) = (b"key".to_vec(), b"value".to_vec());
+
+    // Raw KV
```
Comment (BusyJay) [inline tests/raftstore/test_service.rs]: Can you split it into several small test cases?

Comment (siddontang) [reply tests/raftstore/test_service.rs]: +1

**Comment by BusyJay** [comment]
@disksing PTAL

**Comment by disksing** [comment]
LGTM

File: tests/raftstore/test_service.rs [hunk 2]
```diff
@@ -0,0 +1,471 @@
...
+    del_req.set_context(ctx.clone());
+    del_req.start_key = b"a".to_vec();
+    del_req.end_key = b"z".to_vec();
+    let del_resp = client.kv_delete_range(del_req).unwrap();
+    assert!(!del_resp.has_region_error());
+    assert!(del_resp.error.is_empty());
+}
+
+#[test]
+fn test_raft_and_coprocessor() {
```
Comment (siddontang) [inline tests/raftstore/test_service.rs]: split this test into two.

File: tests/raftstore/test_service.rs [hunk 3]
```diff
@@ -448,24 +448,25 @@ fn test_mvcc_resolve_lock_gc_and_delete() {
...
-            .wait()
-            .unwrap();
-
-        let (sink, _) = client.snapshot();
-        sink.send((SnapshotChunk::new(), Default::default()))
-            .wait()
-            .unwrap();
-    }
+    let (sink, _) = client.raft();
+    sink.send((RaftMessage::new(), Default::default()))
```
Comment (siddontang) [inline tests/raftstore/test_service.rs]: We will not allow region 0 later. So here we need to send an invalid region.

Comment (overvenus) [reply tests/raftstore/test_service.rs]: It is an **invalid** reigon when we do not allow region 0.

Comment (disksing) [reply tests/raftstore/test_service.rs]: @siddontang The grpc service won't return any errors no matter what you send. It just redirect the messages to raftstore.

**Pull request merged**